### PR TITLE
rm newline from end of Markdown text/plain output

### LINKED ISSUE #27102: extra newline displaying markdown objects in REPL
When getting help for a function, or equivalently just displaying a Markdown object, there are now two newlines at the end of the output.

E.g. 0.6:
```
help?> Core.eval
  eval([m::Module], expr::Expr)

  Evaluate an expression in the given module and return the result. Every Module (except those defined with baremodule) has its
  own 1-argument definition of eval, which evaluates expressions in that module.

julia> 
```

and now:
```
help?> Core.eval
  eval([m::Module], expr::Expr)

  Evaluate an expression in the given module and return the result. Every Module (except those defined with baremodule) has its own
  1-argument definition of eval, which evaluates expressions in that module.


julia> 
```

### PR DESCRIPTION
**Pull request by stevengj**
Fixes #27102.

Since I was going through Markdown/render.jl line by line, I also made a few other small cleanups, primarily replacing single-character strings with chars.

### CODE REVIEW

File: stdlib/Markdown/src/render/terminal/render.jl [hunk 1]
```diff
@@ -9,24 +9,26 @@ function term(io::IO, content::Vector, cols)
     isempty(content) && return
     for md in content[1:end-1]
         term(io, md, cols)
-        println(io)
+        print(io, '\n', '\n')
```
Comment (JeffBezanson) [inline stdlib/Markdown/src/render/terminal/render.jl]: Why two newllines?

Comment (JeffBezanson) [reply stdlib/Markdown/src/render/terminal/render.jl]: Oh I see --- because each element used to end with a newline, but those are now removed?

Comment (stevengj) [reply stdlib/Markdown/src/render/terminal/render.jl]: Yes.

File: stdlib/Markdown/src/GitHub/table.jl [hunk 2]
```diff
@@ -142,11 +142,11 @@ function term(io::IO, md::Table, columns)
     padcells!(cells, md.align, len = ansi_length)
     for i = 1:length(cells)
         join(io, cells[i], " ")
-        println(io)
         if i == 1
-            join(io, ["–"^ansi_length(cells[i][j]) for j = 1:length(cells[1])], " ")
             println(io)
+            join(io, ["–"^ansi_length(cells[i][j]) for j = 1:length(cells[1])], " ")
```
Comment (ararslan) [inline stdlib/Markdown/src/GitHub/table.jl]: Would it be worth using `'-'` here as you've done in other places instead of `"-"`?

Comment (stevengj) [reply stdlib/Markdown/src/GitHub/table.jl]: Probably.   There is a lot of this idiom in the Markdown module, however, and my inclination would be to leave it to a separate PR.

Comment (JeffBezanson) [reply stdlib/Markdown/src/GitHub/table.jl]: That would be fine, but I don't think 1-character strings are really so bad.

Comment (stevengj) [reply stdlib/Markdown/src/GitHub/table.jl]: Yes, they are a lot cheaper than a couple of versions ago.   And Markdown rendering is not generally performance-critical.

**Review by fredrikekre — APPROVED** [review]

**Pull request merged**